src/odemis/driver/xt_client.py

```diff
@@ -1506,35 +1536,44 @@ def _acquire(self):
                 while True:
                     if self._acq_should_stop():
                         break
-                    self.parent.set_channel_state(self._channel_name, True)
-                    if self.parent._scanner.blanker.value is None:
-                        self.parent.unblank_beam()
+                    self.parent.set_channel_state(self._scanner.channel, True)
+                    if hasattr(self._scanner, "blanker"):
+                        # TODO In case of switching the scanner type the previous scanner should be blanked.
+                        #  Currently this kind of blanker handling isn't supported by XT.
+                        if self._scanner.blanker.value is None:
+                            self.parent.unblank_beam()
                     # The channel needs to be stopped to acquire an image, therefore immediately stop the channel.
-                    self.parent.set_channel_state(self._channel_name, False)
+                    self.parent.set_channel_state(self._scanner.channel, False)
+                    md = self._scanner._metadata.copy()
 
                     # Estimated time for an acquisition is the dwell time times the total amount of pixels in the image.
-                    n_pixels = self.parent._scanner.shape[0] * self.parent._scanner.shape[1]
-                    est_acq_time = self.parent._scanner.dwellTime.value * n_pixels
+                    if hasattr(self._scanner, "dwellTime") and hasattr(self._scanner, "shape"):
+                        n_pixels = self._scanner.resolution.value[0] * self._scanner.resolution.value[1]
+                        est_acq_time = self._scanner.dwellTime.value * n_pixels
+                        md[model.MD_DWELL_TIME] = self._scanner.dwellTime.value
+                    else:
+                        # Estimated acquisition time used the timeout if not all info about this acquisition time is
+                        # known.
+                        est_acq_time = 60
```

Comment:
There are cases where an acquisition could take several minutes. It'd be a pity to fail consistently on such high level acquisition. So let's raise this timeout to 5*60 s